moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -578,29 +578,29 @@ void IKFastKinematicsPlugin::getSolution(const IkSolutionList<IkReal>& solutions
 
 double IKFastKinematicsPlugin::harmonize(const std::vector<double>& ik_seed_state, std::vector<double>& solution) const
 {
-  double dist_sqr = 0;
+  double dist_abs = 0;
   std::vector<double> ss = ik_seed_state;
   for (size_t i = 0; i < ik_seed_state.size(); ++i)
   {
-    while (ss[i] > 2 * M_PI)
+    while (ss[i] > M_PI)
     {
       ss[i] -= 2 * M_PI;
     }
-    while (ss[i] < 2 * M_PI)
+    while (ss[i] < -1 * M_PI)
     {
       ss[i] += 2 * M_PI;
     }
-    while (solution[i] > 2 * M_PI)
+    while (solution[i] > M_PI)
     {
       solution[i] -= 2 * M_PI;
     }
-    while (solution[i] < 2 * M_PI)
+    while (solution[i] < -1 * M_PI)
     {
       solution[i] += 2 * M_PI;
     }
-    dist_sqr += fabs(ik_seed_state[i] - solution[i]);
+    dist_abs += fabs(ss[i] - solution[i]);
   }
-  return dist_sqr;
+  return dist_abs;
```

Comment:
This is exactly what I am talking about. It had wrong definition. Now it makes more sense and I have also added a [new PR](https://github.com/ros-planning/moveit_kinematics_tests/pull/10) which tests for the getPositionIK call for closest solution to seed. So, Indirectly this test is testing the harmonize function as well. But adding an explicit test for this can be helpful.